draft-ietf-rats-architecture.md

```diff
@@ -897,3 +899,210 @@ Thomas Hardjono created older versions of the terminology section in collaborati
 Eric Voit provided the conceptual separation between Attestation Provision Flows and Attestation Evidence Flows.
 Monty Wisemen created the content structure of the first three architecture drafts.
 Carsten Bormann provided many of the motivational building blocks with respect to the Internet Threat Model.
+
+# Appendix A: Time Considerations {#time-considerations}
+
+The table below defines a number of relevant events, with an ID that
+is used in subsequent diagrams.  The times of said events might be
+defined in terms of an absolute clock time such as Coordinated Universal Time, 
+or might be defined relative to some other timestamp or timeticks counter.
+
+| ID | Event                       | Explanation of event
+|----|-----------------------------|-----------------------
+| VG | Value generation            | A value to appear in a claim was created
+| VN | Nonce sent by Verifier      | A random number not predictable to an Attester is sent by a Verifier
+| VR | Nonce sent by Relying Party | The nonce is sent by a Relying Party
+| EG | Evidence generation         | An Attester collects claims and generates Evidence
+| ER | Evidence relayed            | A Relying Party relays Evidence to a Verifier
+| RG | Result generation           | A Verifier appraises Evidence and generates an Attestation Result
+| RR | Result relayed              | A Relying Party relays an Attestation Result to a Relying Party
+| RA | Result appraised            | The Relying Party appraises Attestation Results
+| OP | Operation performed         | The Relying Party performs some operation requested by the Attester.  For example, acting upon some message just received across a session created earlier at time(RA).
```

Comment:
In a hybrid use case involving multiple Relying Parties, there will be a case where the first RP may issue a result expiry (RX) that is different from the RX supplied by the Verifier. Maybe it is better name it (VX) for Verifier use and allow (RX) for use by Relying Parties?

Reply:
See earlier response about not adding rows unless/until there is text explaining what security test it's required for.  Feel free to suggest text explaining what security threat is relevant and what check some party would do to mitigate it.

Reply:
According to Ned's assumption, here is the case: Verifier -> RP1 -> RP2. The verifier gives a result expiry time in the Attestation Result to RP1. RP2 gives another result expiry to RP2. So RP2 has 2 result expiry times. Maybe Ned wants to emphasize that RP2 has to make a choice between these two times.
But RP can also have other expiry policies that come from other sources, e.g., the administrator configures an expiry time. I think it's up to the implementation to consider how to make decisions among all the policies. What is important here, however, is that the Attestation Result can contain an ‘expiry’ claim.

Reply:
I think you meant "RP1 gives another result expiry to RP2". I suppose it isn't clear whether RP1 can supply RX or if it only can supply RR. If RX is only a function of a Verifier then the use case doesn't make sense. If RX can be asserted by RP to other RPs, then that implies RX can be a "Remediation action" from a RP; which we said might be OK for TEEP arch to define but is likely out of scope for RATS. I prefer to place limits on "who can say what" to limit combinatorics and therefore limits the potential for insecure states. Dave T. seems to be saying that unless there is a use case, it doesn't make sense to define the freshness variable. But even with already defined variables it is possible to find insecure states unless some other constraints are acknowledged. Maybe both perspectives can be held simultaneously? 

Even if we limit RX to Verifier only. An (out-of-scope?) RP1 to RP2 flow could define different RX (RX2) that leaves RP2 in the same undecidability state that Pan Wei points out. But I don't think the arch can prevent the possibility for unsecure states. Nevertheless, I'm still an advocate for specifying which entity is expected to make which freshness claim. Put the burden of proof on the use case to justify when it makes sense for other entities to make the freshness claim.